internal/keyrotation/rotate_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package keyrotation implements the API handlers for running key rotation jobs.
+package keyrotation
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/exposure-notifications-server/internal/database"
+	"github.com/google/exposure-notifications-server/internal/revision"
+	"github.com/google/exposure-notifications-server/internal/serverenv"
+	"github.com/google/exposure-notifications-server/pkg/keys"
+	"github.com/jackc/pgx/v4"
+)
+
+func TestRotateKeys(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	testDB := database.NewTestDatabase(t)
+	kms, _ := keys.NewInMemory(context.Background())
+	env := serverenv.New(ctx, serverenv.WithKeyManager(kms), serverenv.WithDatabase(testDB))
+
+	testCases := []struct {
+		name           string
+		expectKeyCount int64
+	}{
+		{
+			name:           "empty_db",
+			expectKeyCount: 1,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			keyID := "test" + t.Name()
```

Comment:
`t.Parallel()`

Reply:
I was worried about that since these are sharing the database and does a blind write to clear at the end of the test. Need to consider parallel safety.

This test needs work!

Reply:
You can move the database creation into the parallelization too